src/main/java/edu/hm/hafner/analysis/parser/Armcc5CompilerParser.java

```diff
@@ -35,22 +35,22 @@ protected boolean isLineInteresting(final String line) {
 
     @Override
     protected Optional<Issue> createIssue(final Matcher matcher, final LookaheadStream lookahead,
-            final IssueBuilder builder) {
-        var type = matcher.group(3);
-        Severity priority;
-
-        if (equalsIgnoreCase(type, "error")) {
-            priority = Severity.WARNING_HIGH;
-        }
-        else {
-            priority = Severity.WARNING_NORMAL;
-        }
-
-        var errorCode = matcher.group(4);
-        var message = matcher.group(5);
-        return builder.setFileName(matcher.group(1))
-                .setLineStart(matcher.group(2))
+                                          final IssueBuilder builder) {
+        boolean isNewFormat = matcher.group(1) != null;
+
+        String fileName = isNewFormat ? matcher.group(1) : matcher.group(6);
+        String lineStr  = isNewFormat ? matcher.group(2) : matcher.group(7);
+        String type     = isNewFormat ? matcher.group(3) : matcher.group(8);
+        Severity priority = equalsIgnoreCase(type, "error")
+                ? Severity.WARNING_HIGH
+                : Severity.WARNING_NORMAL;
+        String errorCode = isNewFormat ? matcher.group(4) : matcher.group(9);
+        String message   = isNewFormat ? matcher.group(5) : matcher.group(10);
```

Comment:
Can you change this style to something like 
```
if (matcher.group("...") == null) {
   builder.setFileName(matcher.group("..."));
   builder.setSeverity(matcher.group("..."));
else {
   builder.setFileName(matcher.group("..."));
   builder.setSeverity(matcher.group("..."));
}
```

There is no need for variables.